### PR TITLE
add hash_extra

### PR DESCRIPTION
**Pull request by yomi-bazuaye-craft**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -114,6 +114,12 @@ variable "docker_cmd" {
   default     = []
 }
 
+variable "hash_extra" {
+  description = "The string to add into hashing function. Useful when building same source path for multiple functions."
```
Comment (wisrob) [inline variables.tf:118]: It should be included in the README. There pre-commit responsible for the generation. Please execute it.

File: variables.tf [hunk 2]
```diff
@@ -114,6 +114,12 @@ variable "docker_cmd" {
   default     = []
 }
 
+variable "hash_extra" {
```
Comment (wisrob) [inline variables.tf:117]: How is the hash calculated? Maybe we could calculate it in this module based on the source content?

**Review by wisrob — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 2 above)
Comment (adys) [reply variables.tf:117]: @wisrob I'd rather leave the underlying module that we use to handle it: https://github.com/terraform-aws-modules/terraform-aws-lambda . What use-cases do you have in mind?

Comment (yomi-bazuaye-craft) [reply variables.tf:117]: The hash calculation is performed by the AWS Lambda module, see:
https://registry.terraform.io/modules/terraform-aws-modules/lambda/aws/latest#a-namebuild-packagea-how-does-building-and-packaging-work

File: variables.tf (hunk 1 above)
Comment (yomi-bazuaye-craft) [reply variables.tf:118]: Hey @wisrob Could you please rephrase this comment?
I am not sure what you are asking for ...

Comment (yomi-bazuaye-craft) [reply variables.tf:118]: Ah, OK, I get it now!

I guess this:

`There pre-commit responsible for the generation.`

Means:

`The pre-commit is responsible for the generation.`


.. OK, I will make that change now.

File: aws_lambda_function.tf [hunk 3]
```diff
@@ -51,4 +51,5 @@ module "lambda_function" {
 
   use_existing_cloudwatch_log_group = true
   depends_on                        = [aws_cloudwatch_log_group.lambda_log_group]
+  hash_extra                        = var.hash_extra
```
Comment (itallmakesense) [inline aws_lambda_function.tf:54]: why it's not added to variables.tf?

Comment (adys) [reply aws_lambda_function.tf:54]: @yomi-bazuaye-craft you had it in variables.tf I think. Did you remove it?

Comment (yomi-bazuaye-craft) [reply aws_lambda_function.tf:54]: Whoops! Removed this by mistake.
reintroduced now ...

**Review by wisrob — APPROVED** [review]

**Review by adys — APPROVED** [review]

**Pull request merged**